src/shogun/machine/Machine.h

```diff
@@ -237,6 +237,32 @@ class Machine : public StoppableSGObject
 			return true;
 		}
 
+		/** train classifier
+		 *
+		 * @param feat training data
+		 * @param lab training labels
+		 *
+		 * @return pointer to current Machine object
+		 */
+		virtual std::shared_ptr<Machine>
+		fit(std::shared_ptr<Features> feat, std::shared_ptr<Labels> lab)
```

Comment:
```suggestion
		fit(const std::shared_ptr<Features>& feat, const std::shared_ptr<Labels>& lab)
```
?

Reply:
Should these be pure virtual eventually?

Reply:
if we plan to change all the train/apply to fit/predict, then these should be pure virtual, but how do we handle non-parametric models?

Reply:
Non-parametric models would still have to be `fit` right? Because you stil need to store the "train" data? That's why maybe it would be good to subclass Machine to Parametric/NonParametric, where the latter stores features and labels

Reply:
and then ParametricMachine never stores features/labels